app/controllers/api/messaging.php

```diff
@@ -1602,14 +1603,183 @@
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('title', '', new Text(256), 'Title for push notification.')
+    ->param('body', '', new Text(64230), 'Body for push notification.')
+    ->param('data', null, new JSON(), 'Additional Data for push notification.', true)
+    ->param('action', '', new Text(256), 'Action for push notification.', true)
+    ->param('icon', '', new Text(256), 'Icon for push notification.', true)
+    ->param('sound', '', new Text(256), 'Sound for push notification.', true)
+    ->param('color', '', new Text(256), 'Color for push notification.', true)
+    ->param('tag', '', new Text(256), 'Tag for push notification.', true)
+    ->param('badge', '', new Text(256), 'Badge for push notification.', true)
+    ->param('status', 'processing', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or processing.', true)
+    ->param('deliveryTime', null, new DatetimeValidator(requireDateInFuture: true), 'Delivery time for message in ISO 8601 format. DateTime value must be in future.', true)
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('messaging')
+    ->inject('response')
+    ->action(function (string $messageId, string $providerId, array $to, string $description, string $title, string $body, ?array $data, string $action, string $icon, string $sound, string $color, string $tag, string $badge, string $status, ?string $deliveryTime, Database $dbForProject, Document $project, Messaging $messaging, Response $response) {
+        $messageId = $messageId == 'unique()' ? ID::unique() : $messageId;
+
+        $provider = $dbForProject->getDocument('providers', $providerId);
+
+        if ($provider->isEmpty() || $provider->getAttribute('type') !== 'push') {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        $pushData = [
+            'title' => $title,
+            'body' => $body,
+        ];
+
+        if (!is_null($data)) {
+            $pushData['data'] = $data;
+        }
+
+        if ($action) {
+            $pushData['action'] = $action;
+        }
+
+        if ($icon) {
+            $pushData['icon'] = $icon;
+        }
+
+        if ($sound) {
+            $pushData['sound'] = $sound;
+        }
+
+        if ($color) {
+            $pushData['color'] = $color;
+        }
+
+        if ($tag) {
+            $pushData['tag'] = $tag;
+        }
+
+        if ($badge) {
+            $pushData['badge'] = $badge;
+        }
+
+        $message = $dbForProject->createDocument('messages', new Document([
+            '$id' => $messageId,
+            'providerId' => $provider->getId(),
+            'providerInternalId' => $provider->getInternalId(),
+            'to' => $to,
+            'description' => $description,
+            'data' => $pushData,
+            'status' => $status,
+            'search' => $messageId . ' ' . $description . ' ' . $title . ' ' . $providerId,
```

Comment:
Let's make sure we set deliveryTime here

Reply:
done this